# Ôn tập

Notebook này dùng để làm quen với một số thao tác cơ bản để thao tác trên dữ liệu sử dụng thư viện `Pandas`. Dữ liệu sử dụng trong notebook này có kích thước rất nhỏ để các bạn có thể dễ kiểm tra kết quả viết mã của mình. Nhằm mục đích phát triển khả năng tự học, tự tìm kiếm câu trả lời, các câu hỏi sẽ không có lời giải mà chỉ cung cấp đáp án ở một số câu để các bạn có thể tự đối chiếu. Các bạn có thể tự tìm kiếm câu trả lời từ nhiều nguồn khác nhau, mỗi câu hỏi có thể sẽ có nhiều cách giải quyết khác nhau.

Thông tin các bộ dữ liệu:

- `car-sales.csv`: bộ dữ liệu chứa thông tin của các ôtô cũ ở một cửa hàng


### 1. Đọc dữ liệu

In [ ]:
# Your code here

import pandas as pd
df = pd.read_csv('/content/sample_data/car-sales.csv')
print(df)



  Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0


### 2. Dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [ ]:
# Your code here
# Đọc dữ liệu từ tệp CSV
import pandas as pd
data = pd.read_csv('/content/sample_data/car-sales.csv')

# Sử dụng .shape để lấy số dòng và số cột
num_rows, num_columns = data.shape

# In ra số dòng và số cột
print(f"Số dòng: {num_rows}")
print(f"Số cột: {num_columns}")


Số dòng: 10
Số cột: 5


### 3. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [ ]:
# Your code here
import pandas as pd

# Đọc dữ liệu từ tệp CSV
data = pd.read_csv('/content/sample_data/car-sales.csv')

# Tạo danh sách các thuộc tính phân loại và định lượng
categorical_attributes = []
numeric_attributes = []

# Xác định thuộc tính phân loại và định lượng
for column in data.columns:
    if data[column].dtype == "object":
        categorical_attributes.append(column)
    else:
        numeric_attributes.append(column)

# In danh sách thuộc tính
print("Thuộc tính phân loại:")
for cat_attr in categorical_attributes:
    print(cat_attr)

print("\nThuộc tính định lượng:")
for num_attr in numeric_attributes:
    print(num_attr)


Thuộc tính phân loại:
Manufacturer
Colour

Thuộc tính định lượng:
Odometer
Doors
Price


### 4. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [ ]:
# Your code here
import pandas as pd

# Đọc dữ liệu từ tệp CSV
data = pd.read_csv('/content/sample_data/car-sales.csv')

# Sử dụng .shape để lấy số dòng (số xe) trong dữ liệu
so_xe = data.shape[0]

# In ra số lượng xe
print(f"Trong dữ liệu có thông tin về {so_xe} xe.")


Trong dữ liệu có thông tin về 10 xe.


### 5. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [ ]:
# Your code here
import pandas as pd

# Đọc dữ liệu từ tệp CSV
data = pd.read_csv('/content/sample_data/car-sales.csv')

# Sử dụng .value_counts() để đếm số xe theo hãng (Manufacturer)
so_xe_theo_hang = data["Manufacturer"].value_counts()

# In ra kết quả
print("Số xe theo hãng:")
print(so_xe_theo_hang)


Số xe theo hãng:
Toyota    4
Honda     3
Nissan    2
BMW       1
Name: Manufacturer, dtype: int64


### 6. Đếm số xe theo màu


In [ ]:
# Your code here
import pandas as pd

# Đọc dữ liệu từ tệp CSV
data = pd.read_csv('/content/sample_data/car-sales.csv')

# Sử dụng .value_counts() để đếm số xe theo màu sắc
so_xe_theo_mau = data["Colour"].value_counts()

# In ra kết quả
print("Số xe theo màu sắc:")
print(so_xe_theo_mau)


Số xe theo màu sắc:
White    4
Blue     3
Red      1
Black    1
Green    1
Name: Colour, dtype: int64


### 7. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [ ]:
# Your code here
import pandas as pd

# Đọc dữ liệu từ tệp CSV
data = pd.read_csv('/content/sample_data/car-sales.csv')

# Sử dụng .sum() để tính tổng giá trị của tất cả các xe
tong_gia_tri_xe = data["Price"].sum()

# In ra kết quả
print(f"Tổng giá trị của tất cả các xe trong cửa hàng là: ${tong_gia_tri_xe}")


Tổng giá trị của tất cả các xe trong cửa hàng là: $76450.0


### 8. Top 3 xe có giá cao nhất


In [ ]:
# Your code here
import pandas as pd

# Đọc dữ liệu từ tệp CSV
data = pd.read_csv('/content/sample_data/car-sales.csv')

# Sắp xếp dữ liệu theo cột "Price" theo thứ tự giảm dần và chọn 3 xe đầu tiên
top_3_xe_gia_cao_nhat = data.sort_values(by="Price", ascending=False).head(3)

# In ra kết quả
print("Top 3 xe có giá cao nhất:")
print(top_3_xe_gia_cao_nhat)


Top 3 xe có giá cao nhất:
  Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0


### 9. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [ ]:
# Your code here
import pandas as pd

# Đọc dữ liệu từ tệp CSV
data = pd.read_csv('/content/sample_data/car-sales.csv')

# Sắp xếp dữ liệu theo cột "Odometer" theo thứ tự giảm dần và chọn 3 xe đầu tiên
top_3_xe_odometer_cao_nhat = data.sort_values(by="Odometer", ascending=False).head(3)

# In ra kết quả
print("Top 3 xe đã chạy nhiều cây số nhất (Odometer):")
print(top_3_xe_odometer_cao_nhat)



Top 3 xe đã chạy nhiều cây số nhất (Odometer):
  Manufacturer Colour  Odometer  Doors   Price
4       Nissan  White    213095      4  3500.0
0       Toyota  White    150043      4  4000.0
5       Toyota  Green     99213      4  4500.0


### 10. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [ ]:
# Your code here
import pandas as pd

# Đọc dữ liệu từ tệp CSV
data = pd.read_csv('/content/sample_data/car-sales.csv')

# 1. Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng
ty_le_honda = len(data[data["Manufacturer"] == "Honda"]) / len(data)
print(f"Tỷ lệ xe Honda trong dữ liệu của cửa hàng: {ty_le_honda:.1%}")

# 2. Tính giá trung bình của xe hãng Honda của cửa hàng
gia_trung_binh_honda = data[data["Manufacturer"] == "Honda"]["Price"].mean()
print(f"Giá trung bình của xe hãng Honda của cửa hàng: ${gia_trung_binh_honda:.2f}")

# 3. Liệt kê các màu sắc của xe Honda trong cửa hàng
mau_sac_honda = data[data["Manufacturer"] == "Honda"]["Colour"].unique()
print("Các màu sắc của xe Honda trong cửa hàng:", ", ".join(mau_sac_honda))


Tỷ lệ xe Honda trong dữ liệu của cửa hàng: 30.0%
Giá trung bình của xe hãng Honda của cửa hàng: $6500.00
Các màu sắc của xe Honda trong cửa hàng: Red, Blue


# 11. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [ ]:
# Your code here
import pandas as pd

# Đọc dữ liệu từ tệp CSV
data = pd.read_csv('/content/sample_data/car-sales.csv')

# Tính giá trung bình của xe
gia_trung_binh = data["Price"].mean()
print(f"Giá trung bình của xe: ${gia_trung_binh:.2f}")

# Tính giá trị trung vị của xe
gia_tri_trung_vi = data["Price"].median()
print(f"Giá trị trung vị của xe: ${gia_tri_trung_vi:.2f}")

# Tính miền giá trị của xe
gia_min = data["Price"].min()
gia_max = data["Price"].max()
mien_gia_tri = gia_max - gia_min
print(f"Miền giá trị của xe: ${mien_gia_tri}")


# Tính miền phân vị của xe (IQR)
q1 = data["Price"].quantile(0.25)
q3 = data["Price"].quantile(0.75)
iqr = q3 - q1
print(f"Miền phân vị của xe (IQR): ${iqr}")

# Tính điểm phân vị 0.4 và 0.6 của giá
percentile_40 = data["Price"].quantile(0.4)
percentile_60 = data["Price"].quantile(0.6)
print(f"Điểm phân vị 0.4 và 0.6 của giá: [${percentile_40}, ${percentile_60}]")



Giá trung bình của xe: $7645.00
Giá trị trung vị của xe: $6625.00
Miền giá trị của xe: $18500.0
Miền phân vị của xe (IQR): $2750.0
Điểm phân vị 0.4 và 0.6 của giá: [$5750.0, $7000.0]


---